# 1.requests 모듈 : HTTP프로토콜을 사용할 수 있게 해주는 모듈
* requsest모듈 사용 설명서
[requests](https://requests.readthedocs.io/en/latest/user/quickstart/)

# 2.requests 모듈 설치
* conda install requests
* pip install requests

# 3. requests 모듈 사용법

* 1) 모듈 불러오기: import requests
* 2) url 변수에 저장
* 3) url 파라미터
  * payload = {파라미터1 : value, 파라미터2 : value}
* 4) headers
  * headers = {key : value}
* 5) HTTP 요청 보내기 get, post
  * get: r = requests.get(url, params=payload, headers=headers)
  * post: r = requests.post(url, data={key:value})
* 6) 만들어진 url 확인: print(r.url)
* 7) 응답 코드 확인: print(r.status_code)
* 8) 응답 요소 출력: r.text, r.content, r.json()

In [ ]:
import requests
url =""
payload = dict()
r = requests.get(url, params=payload)
print(r.url)
print(r.status_code)
response = r.text # r.content, r.json()

# requests 모듈로 1페이지만 가져오기

In [ ]:
url = "https://openapi.naver.com/v1/search/book.json"
Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
Client_Secret = "by6xHZcrlB"
payload = dict(query="핀테크", display=100, start=start_num, sort="sim")
headers = {"X-Naver-Client-Id": Client_Id,
           "X-Naver-Client-Secret": Client_Secret}
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
response = r.json()

In [27]:
start_num2 = 1
for s_num in range(11):
    if start_num2 < 901:
        print(start_num2)
        start_num2 += 100
    elif start_num2 == 901:
        print(start_num2)
        start_num2 += 99
        print(start_num2)

1
101
201
301
401
501
601
701
801
901
1000


In [33]:
import requests
start_num = 1
result_json = []
for page in range(11):
    url = "https://openapi.naver.com/v1/search/book.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    payload = dict(query="핀테크", display=100, start=start_num, sort="sim")
    headers = {"X-Naver-Client-Id": Client_Id,
               "X-Naver-Client-Secret": Client_Secret}
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    response = r.json()
    result_json.append(response)
    if start_num < 901:
        start_num += 100
    elif start_num == 901:
        start_num += 99

https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=201&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=301&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=401&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=501&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=601&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=701&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&di

* 모든 검색 결과를 가져오기 위해서는 여러 페이지 정보를 수집
* 전체 결과 / 1페이지 표시 개수 + 1 = 전체 페이지수
* start 부분 + 1페이지 표시 개수 = 다음 페이지 데이터 수집 가능
* 반복문을 이용해서 start 에 100씩 더해서 최대 1000이 되도록 하면 최대한 많은 데이터 수집 가능

In [50]:
import requests
url = "https://openapi.naver.com/v1/search/book.json"
Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
Client_Secret = "by6xHZcrlB"
payload = dict(query="핀테크", display=100, start=1, sort="sim")
headers = {"X-Naver-Client-Id": Client_Id,
           "X-Naver-Client-Secret": Client_Secret}
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
response = r.json()


https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200


In [51]:
pages = response['total'] // response['display'] + 1
pages

2

In [52]:
start_num = 1
response_json2 = []
for page in range(pages):
    url = "https://openapi.naver.com/v1/search/book.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    payload = dict(query="핀테크", display=100, start=start_num, sort="sim")
    headers = {"X-Naver-Client-Id": Client_Id,
               "X-Naver-Client-Secret": Client_Secret}
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    response = r.json()
    pages = response['total'] // response['display'] + 1
    response_json2.append(response)
    if start_num == 1000:
        break
    if start_num < 901:
        start_num += 100
    elif start_num == 901:
        start_num += 99

https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200


In [54]:
response_json2[1]

{'lastBuildDate': 'Tue, 01 Apr 2025 11:28:31 +0900',
 'total': 106,
 'start': 101,
 'display': 6,
 'items': [{'title': '핀테크 / 블록체인 기술이 적용된 온라인 시스템 전자결제 / 보안 / 인증 시장 전망과 핵심기술 동향',
   'link': 'https://search.shopping.naver.com/book/catalog/32438161048',
   'image': 'https://shopping-phinf.pstatic.net/main_3243816/32438161048.20221227205547.jpg',
   'author': 'TF정보분석실',
   'discount': '342000',
   'publisher': '트렌드포커스',
   'pubdate': '20190429',
   'isbn': '9791188774074',
   'description': '4차 산업혁명 시대에 핀테크와 블록체인이 핵심기술로 떠오르고 있습니다.\n핀테크는 금융서비스 가치와 혁신을 향상시킬 수 있는 비즈니스 모델로 대두되고 있으며 \n블록체인은 세계경제포럼(World Economic Forum, WEF)에서 4차 산업혁명 시대의 핵심\n기술로 선정될 정도로 인정받고 있고 올해까지 전 세계 은행의 80%가 블록체인 기술을 도\n입할 것으로 전망되고 있습니다.\n\n핀테크(fintech)는 금융서비스 도메인에서 IT 기술을 기반으로 새로운 응용프로그램, 프로\n세스, 제품, 비즈니스 모델 또는 새로운 서비스를 제공하는 것으로 이를 통해 금융서비스의 \n가치와 혁신을 향상시키고 있으며 IT기술의 발전에 따라 간편 결제, 대출 프레임워크, 크라\n우드 펀딩등 다양한 형태의 금융서비스가 출시되고 있습니다.\n그리고 블록체인은 금융 분야를 비롯하여 제조 및 유통, 공공서비스, 사회?문화 부문의 \n전영역에 걸쳐 혁신적인 변화를 일으키는 새로운 패러다임이라 말할 수 있습니다.\

# While True로 전체데이터 가져오기

In [56]:
page = 1
start_num2 = 1
response_json3 = []
while True:
    url = "https://openapi.naver.com/v1/search/blog.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    payload = dict(query="핀테크", display=100, start=start_num2, sort="sim")
    headers = {"X-Naver-Client-Id": Client_Id,
               "X-Naver-Client-Secret": Client_Secret}
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    response = r.json()
    total_pages = (response['total'] // 100) + 1
    response_json3.append(response)
    if start_num2 == 1000:
        break
    if start_num2 < 901:
        start_num2 += 100
    elif start_num2 == 901:
        start_num2 += 99
    
    if page < total_pages:
        page += 1
    else:
        break
print(len(response_json3))

https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=201&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=301&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=401&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=501&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=601&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=701&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&di

# 블로그, 책, 뉴스 동시에 수집하기

In [59]:
import time
response_json3 = []
for category in ["blog", "news", "book"]:
    page = 1
    start_num2 = 1
    while True:
        url = f"https://openapi.naver.com/v1/search/{category}.json"
        Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
        Client_Secret = "by6xHZcrlB"
        payload = dict(query="핀테크", display=100, start=start_num2, sort="sim")
        headers = {"X-Naver-Client-Id": Client_Id,
                   "X-Naver-Client-Secret": Client_Secret}
        r = requests.get(url, params=payload, headers=headers)
        print(r.url)
        print(r.status_code)
        response = r.json()
        total_pages = (response['total'] // 100) + 1
        response_json3.append(response)
        if start_num2 == 1000:
            break
        if start_num2 < 901:
            start_num2 += 100
        elif start_num2 == 901:
            start_num2 += 99

        if page < total_pages:
            page += 1
        else:
            break
        time.sleep(0.5)
print(len(response_json3))

https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=201&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=301&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=401&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=501&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=601&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=701&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&di

# 블로그, 뉴스, 책 데이터 각각 저장하기

In [75]:
import time
keyword = input("검색할 키워드를 입력해 주세요 ")
response_blog = []
response_news = []
response_book = []
for category in ["blog", "news", "book"]:
    page = 1
    start_num2 = 1
    while True:
        url = f"https://openapi.naver.com/v1/search/{category}.json"
        Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
        Client_Secret = "by6xHZcrlB"
        payload = dict(query=keyword, display=100, start=start_num2, sort="sim")
        headers = {"X-Naver-Client-Id": Client_Id,
                   "X-Naver-Client-Secret": Client_Secret}
        r = requests.get(url, params=payload, headers=headers)
        print(r.url, end="\r")
        print(r.status_code, end="\r")
        response = r.json()
        total_pages = (response['total'] // 100) + 1
        
        if category == 'blog':
            response_blog.append(response)
        elif category == "news":
            response_news.append(response)
        elif category == "book":
            response_book.append(response)
                        
        if start_num2 == 1000:
            break
        if start_num2 < 901:
            start_num2 += 100
        elif start_num2 == 901:
            start_num2 += 99

        if page < total_pages:
            page += 1
        else:
            break
        time.sleep(0.5)
print(len(response_blog))
print(len(response_news))
print(len(response_book))

검색할 키워드를 입력해 주세요 인공지능
110ps://openapi.naver.com/v1/search/book.json?query=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&display=100&start=1000&sort=sim
11
11


In [7]:
def json_unpack(all_data):
    final_result = {}
    for page in all_data:
        for item in page['items']:
            for key, value in item.items():
                final_result.setdefault(key, []).append(value)
    df = pd.DataFrame(final_result)    
    return df

In [78]:
for jsons, cat in zip([response_blog, response_news, response_book],["blog", "news", "book"]):
    unpack_result = json_unpack(jsons)
    unpack_result.to_csv(f"naver_{cat}_result.csv", encoding="utf-8", index=False)

# 3개 카테고리 수집 코드를 함수화하기

In [10]:
import requests
import time
import os
import pandas as pd

In [11]:
def naver_search(keyword, category):
    response_result = []
    page = 1
    start_num2 = 1
    while True:
        url = f"https://openapi.naver.com/v1/search/{category}.json"
        Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
        Client_Secret = "by6xHZcrlB"
        payload = dict(query=keyword, display=100, start=start_num2, sort="sim")
        headers = {"X-Naver-Client-Id": Client_Id,
                   "X-Naver-Client-Secret": Client_Secret}
        r = requests.get(url, params=payload, headers=headers)
        print(r.url, end="\r")
        print(r.status_code, end="\r")
        response = r.json()
        total_pages = (response['total'] // 100) + 1
        
        response_result.append(response)
                        
        if start_num2 == 1000:
            break
        if start_num2 < 901:
            start_num2 += 100
        elif start_num2 == 901:
            start_num2 += 99

        if page < total_pages:
            page += 1
        else:
            break
        time.sleep(0.2)
    
    return response_result


In [12]:
import time
keyword = input("검색할 키워드를 입력해 주세요 ")
for category in ["blog", "news", "book"]:
    naver_search_result = naver_search(keyword, category)
    result_df = json_unpack(naver_search_result)
    display(result_df)

검색할 키워드를 입력해 주세요 인공지능


,title,link,description,bloggername,bloggerlink,postdate
0,제주도 1박2일 여행 스케줄 <b>인공지능</b> 프로그램 챗GPT의 도움을...,https://jongamk.tistory.com/526077,<b>인공지능</b> 프로그램 챗GPT로 떠난 제주 1일차 과연 제주도 여행을 챗G...,핑구야 날자의 IT와 일상,https://jongamk.tistory.com/,20250323
1,학점은행제 <b>인공지능</b> 학위 취득 직장인도 할 수 있네요!,https://blog.naver.com/2_dl/223813189505,"등에 인공적으로 구현한, 모방한 기능을 갖춘 컴퓨터 시스템으로 이와 같이 지능을 만...",2_dl님의블로그,blog.naver.com/2_dl,20250328
2,<b>인공지능</b>학과 체계적인 대비방법,https://blog.naver.com/blackhole_chinese/22378...,<b>인공지능</b>학과 합격자의 후기를 소개합니다 <b>인공지능</b>학과 체계적...,묘한 교육 연구소,blog.naver.com/blackhole_chinese,20250306
3,고양이홈캠 도기보기 도기캠 <b>인공지능</b>으로 스마트하게,https://blog.naver.com/sunlovely09/223817969436,<b>인공지능</b> AI 기능도 추가되어 있어서 반려동물을 좀더 세심하게 맞춤형으...,소중한 나날들,blog.naver.com/sunlovely09,20250401
4,<b>인공지능</b> 스피커 종류 KT 기가지니4 스펙 요금제 알아보기(lg...,https://blog.naver.com/minsha321/223816637896,"&quot; &quot; <b>인공지능</b> 스피커, 어떤 제품을 선택해야 좋을까...",일해라 절해라,blog.naver.com/minsha321,20250331
...,...,...,...,...,...,...
1095,<b>인공지능</b>과 사진,https://blog.naver.com/commbooks/223631394459,"<b>인공지능</b>은 손과 카메라를 넘어 이미지를 생성하며, GAN 같은 기술은 ...",커뮤니케이션북스,blog.naver.com/commbooks,20241024
1096,<b>인공지능</b>을 활용해 아이 맞춤형 영어 교재 만들기,https://blog.naver.com/jonathanshim/223742139185,<b>인공지능</b>이 고도로 발전하면서 이제 교육계의 오랜 염원이던 학생 맞춤형 ...,심정섭의 학군과 교육,blog.naver.com/jonathanshim,20250130
1097,"관악구,<b>인공지능</b>(AI) 아동 그림 심리검사서비스",https://blog.naver.com/coreap6/223658753251,"관악구,‘<b>인공지능</b>(AI) 아동 그림 심리검사서비스’로 아동학대 예방 총...",새서울신문,blog.naver.com/coreap6,20241113
1098,AI <b>인공지능</b> 로봇 알파미니(Alpha Mini)로 배우는 <b>인공지...,https://blog.naver.com/mdhb7/223790924778,"알파미니는 소형 <b>인공지능</b> 로봇으로, 27cm의 작은 크기를 가지고 있어...",모두행복,blog.naver.com/mdhb7,20250310


,title,originallink,link,description,pubDate
0,"&quot;AI 에어컨 수요 적극 대응&quot;…삼성전자, 생산라인 풀가동",https://www.etoday.co.kr/news/view/2458159,https://www.etoday.co.kr/news/view/2458159,이투데이=박민웅 기자 | 삼성전자가 올해 무더위에 따른 에어컨 수요에 신속하게 대응...,"Wed, 02 Apr 2025 08:32:00 +0900"
1,애플 AI 한국어 지원 시작했지만…이용자 맞춤형 '시리'는 언제 나와,https://www.newsis.com/view/NISX20250401_00031...,https://n.news.naver.com/mnews/article/003/001...,출시 약 반년 만에 한국어 지원을 시작한 애플의 <b>인공지능</b>(AI) 시스템...,"Wed, 02 Apr 2025 06:01:00 +0900"
2,올트먼 &quot;GPU 녹는다&quot; 행복한 비명…'지브리 프사' 대유행,https://www.hankyung.com/article/202504012315g,https://n.news.naver.com/mnews/article/015/000...,오픈AI가 지난달 25일 선보인 ‘챗GPT-4o’의 새 <b>인공지능</b>(AI)...,"Tue, 01 Apr 2025 10:30:00 +0900"
3,"중앙대 <b>인공지능</b>인문학연구소, '인문한국 3.0 지원사업' 선정",https://edu.donga.com/news/articleView.html?id...,https://edu.donga.com/news/articleView.html?id...,중앙대학교 <b>인공지능</b>인문학연구소가 교육부 산하 한국연구재단이 지원하는 '...,"Tue, 01 Apr 2025 16:54:00 +0900"
4,금목걸이 절도범 <b>인공지능</b> 선별 관제 시스템에 '딱 걸렸네',https://www.news1.kr/local/jeonbuk/5738541,https://n.news.naver.com/mnews/article/421/000...,중고 거래 앱 '당근마켓'을 통해 금목걸이를 건네받은 뒤 오토바이를 타고 도주한 4...,"Tue, 01 Apr 2025 11:23:00 +0900"
...,...,...,...,...,...
1095,경륜·경정 누리집(홈페이지) 전면 재구축 ⋯ <b>인공지능</b> 기반 서비스 강...,https://www.mk.co.kr/article/11259896,https://m.sports.naver.com/general/article/410...,이번 경륜·경정 누리집 개편은 <b>인공지능</b>(AI) 기술을 활용한 순위 예측...,"Tue, 11 Mar 2025 08:01:00 +0900"
1096,"KAI, <b>인공지능</b> 데이터솔루션 기업 젠젠AI 지분 10% 확보",https://biz.sbs.co.kr/article_hub/20000221496?...,https://n.news.naver.com/mnews/article/374/000...,한국항공우주산업(KAI)은 6일 <b>인공지능</b>(AI) 기반 국방 합성 데이터...,"Fri, 07 Mar 2025 10:58:00 +0900"
1097,"한국표준협회, 고누아이와 '<b>인공지능</b> 인재 양성과정' 개최",https://www.enetnews.co.kr/news/articleView.ht...,https://www.enetnews.co.kr/news/articleView.ht...,"한국표준협회(회장 문동민, 협회)는 인텔 코리아, <b>인공지능</b>(AI) 교육...","Tue, 18 Mar 2025 15:12:00 +0900"
1098,"IBK기업은행, '제3회 해군해병대 <b>인공지능</b>(AI) 경진대회' 후원",https://weekly.cnbnews.com/news/article.html?n...,https://weekly.cnbnews.com/news/article.html?n...,사진=IBK기업은행 IBK기업은행(은행장 김성태)이 지난 7일 해군해병대 AI 역량...,"Mon, 10 Mar 2025 14:42:00 +0900"


,title,link,image,author,discount,publisher,pubdate,isbn,description
0,"인공지능 (수학적 이해에서 알고리즘까지, 개정판)",https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_530905...,오상훈,26730,홍릉,20250311,9791156002635,
1,교실 밖 인공지능 수업 (우리의 상상을 미래의 현실로 만들어가는 인공지능의 세계를 ...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_337008...,김현정,12600,궁리,20220725,9788958207788,다가올 새로운 미래를 준비하는 청소년들을 위해 \nIT전문가가 현장에서 들려주는 생...
2,AI 리터러시 (인공지능 필수 지식부터 완벽 활용까지),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_516051...,김용성,22500,프리렉,20241202,9788965404026,AI 시대에 필수로 알아야 할 인공지능의 원리와 개념부터\n학생·직장인·공무원·교육...
3,된다! 하루 만에 끝내는 챗GPT 활용법 (인공지능에게 일 시키고 시간 버는 법),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_493627...,프롬프트 크리에이터,15480,이지스퍼블리싱,20240729,9791163036234,《된다! 하루 만에 끝내는 챗GPT 활용법》이 GPT-4o를 반영한 전면 개정 2판...
4,인공지능 시대의 컴퓨터 개론 (개정8판),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_365674...,김대수^김경동,25760,생능출판,20221227,9788970506173,"제1장에서는 디지털 혁명과 4차 산업혁명의 시대를 다루었다. 인공지능 기술, 디지털..."
...,...,...,...,...,...,...,...,...,...
1095,모두의 인공지능 with 스크래치 (누구나 쉽게 만들면서 배우는 인공지능 기초),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324362...,이영호,16200,길벗,20190918,9791160509052,"개념은 쉽게, 실습은 재밌게! \n남녀노소 누구나 볼 수 있는 실용적인 인공지능 입..."
1096,브리태니커 만화 백과 51: 인공지능,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324363...,봄봄 스토리,9900,미래엔아이세움,20170822,9788937888816,미국에서 인공지능을 연구하는 공 박사는 휴가를 맞아 쌍둥이 조카 슬기와 로운이의 집...
1097,4차 산업혁명과 인공지능 영어교육 (제2판),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324927...,김인석,22500,한국문화사,20210330,9791166850011,이 책은 영어교육을 담당하는 전문가들이 4차 산업혁명 시대 정신에 부합하는 영어교육...
1098,인공지능과 법률 서비스 분야의 혁신,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324776...,양종모,22500,한국학술정보,20210331,9791166033940,"인공지능의 개념과 개발 동향부터, 법률 서비스 영역에서의 인공지능 알고리즘 활용까지..."


In [ ]:
len(naver_search_result)

In [17]:
import requests
url = "https://openapi.naver.com/v1/search/blog"
payload = {'query': '핀테크', 'display' : 100, 'start': 1, 'sort' : "sim"}
headers = {"X-Naver-Client-Id" :"sk2iKdMANqpvG3Pa2WIJ", "X-Naver-Client-Secret" : "by6xHZcrlB"}
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
data = r.json()
data

https://openapi.naver.com/v1/search/blog?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200


{'lastBuildDate': 'Mon, 31 Mar 2025 14:38:57 +0900',
 'total': 456049,
 'start': 1,
 'display': 100,
 'items': [{'title': '<b>핀테크</b> 투자 후기',
   'link': 'https://blog.naver.com/kan1687/223812724931',
   'description': '<b>핀테크</b> 투자, 아직도 망설이고 계신가요?  여러분 혹시 이런 고민 해보신 적 있나요? &quot;<b>핀테크</b> 분야가 떴다는데, 지금 시작해도 괜찮을까? 좀 더 상황을 지켜봐야 하나?&quot; 사실 이런 생각은 투자를 고민하는 우리... ',
   'bloggername': '우정잉♥',
   'bloggerlink': 'blog.naver.com/kan1687',
   'postdate': '20250329'},
  {'title': '[<b>핀테크</b>] 2024년 <b>핀테크</b> 업계 결산: 놀라운 3가지 변화',
   'link': 'https://blog.naver.com/xoom777/223708685044',
   'description': '동안 <b>핀테크</b> 업계에는 예상치 못한 변화의 바람이 불었습니다. 한동안 주춤했던 트렌드가... 올 한 해 <b>핀테크</b> 업계를 뜨겁게 달군 3가지 트렌드에 대해 자세히 살펴보겠습니다. 암호화폐의 귀환, 기관... ',
   'bloggername': 'Ich fange an',
   'bloggerlink': 'blog.naver.com/xoom777',
   'postdate': '20241229'},
  {'title': '[인사이트 캠퍼스] 2025 동계 <b>핀테크</b> 캠프 (with 전주대학교... ',
   'link': 'https://blog.naver.com/fininsight/223734638126',
   'description': '전주대학교 친구들이 직접 여의

In [6]:
payload

{'query': '핀테크', 'display': 100, 'start': 1, 'sort': 'sim'}

In [9]:
headers

{'client_id': 'sk2iKdMANqpvG3Pa2WIJ', 'client_secret': 'by6xHZcrlB'}

네이버 검색 API를 이용해서 책을 검색하고 결과를 데이터프레임으로 만드세요.
* 검색어는 input()을 이용해서 입력받은 값을 이용하세요.

In [19]:
import requests
import pandas as pd

In [4]:
def json_unpack(data):
    result = {}
    for item in data['items']:
        for key, value in item.items():
            result.setdefault(key, []).append(value)
    df = pd.DataFrame(result)
    return df

In [62]:
keyword = input("검색할 키워드를 입력해 주세요: ")

검색할 키워드를 입력해 주세요: 파이썬


In [69]:
url = "https://openapi.naver.com/v1/search/book.json"
Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
Client_Secret = "by6xHZcrlB"
headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
payload = dict(query=keyword, display=100, start=1000, sort="sim")
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
data = r.json()
print("전체 검색 데이터 수: ", data['total'], "검색할 페이지수: ", (int(data['total'])//100)+1)
result_df = json_unpack(data)
result_df

https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=1000&sort=sim
200
전체 검색 데이터 수:  1113 검색할 페이지수:  12


,title,link,image,author,discount,publisher,pubdate,isbn,description
0,파이썬과 케라스를 이용한 딥러닝/강화학습 주식투자 (퀀트 투자 알고리즘 트레이딩을 ...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324920...,김문권,0,위키북스,20180529,9791158391065,강화학습은 스스로 학습하는 머신러닝 기법으로서 주식 데이터 학습에 잘 적용되는 기법...
1,슬기로운 코딩 생활 with 파이썬 으랏차차 기초편 (즐겁게 시작하고 똑똑하게 끝내...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324902...,김정욱,0,루비페이퍼,20180515,9791186710333,"수학, 과학, 국어, 사회 등 실제 초중등 교과 과목과 연계된 예제로 코딩력과 학습..."
2,2024 내일은 빅데이터분석기사 실기 with 파이썬 (제6회부터 변경된 작업형 제...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_434140...,Jade Jeong,20700,김앤북,20231030,9788965127925,「2024 내일은 빅데이터분석기사 실기」는 자격증을 준비하는 수험생 & 재직자(비전...
3,2024 빅데이터분석기사 실기 한권완성: 파이썬(Python) (작업형 제3유형 완...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_461080...,최예신^박진원^이경숙^김주현,27000,예문에듀,20240520,9791163862833,빅데이터분석기사 실기시험 초단기 합격을 위한 최고의 실전 대비서\n‘2024 빅데이...
4,Step by Step CRM을 위한 파이썬 머신러닝 (빅 데이터 분석 자격증 머신...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_394909...,김형수,30000,프레딕스,20230501,9791195511792,"본서는 파이썬을 전혀 모르는 사람도, 프로그래밍을 전혀 해보지 않은 사람도 단계적으..."
...,...,...,...,...,...,...,...,...,...
95,"챗GPT와 업무자동화 (1분 만에 끝내는 엑셀, 구글 시트, 파워포인트, 파이썬 자...",https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_386135...,김철수,16200,위키북스,20230321,9791158394288,"코드 한 줄 몰라도 1분 만에 끝내는 엑셀, 구글 스프레드시트, 파워포인트 업무 자..."
96,프로그래밍 정석(개념을 잡아주는) (기초 연산부터 고급 문제해결 능력까지 파이썬을 ...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324972...,제니퍼 캠벨^폴 그리스^제이슨 몬토조^그렉 윌슨,0,에이콘출판,20100504,9788960771338,"『개념을 잡아주는 프로그래밍 정석』은 기본적인 프로그래밍 요소부터 알고리즘, 디버깅..."
97,머신러닝 딥러닝을 이용한 A.I. 자율주행 RC카 프로젝트 with 파이썬+아두이노...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324665...,서민우,0,앤써북,20220520,9791185553979,〈머신러닝 딥러닝을 이용한 A.I. 자율주행 RC카 프로젝트 with 파이썬+아두이...
98,파이썬 딥러닝 파이토치(Python Deep Learning PyTorch),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324549...,이경택^방성수^안상준,14800,정보문화사,20201008,9788956748573,파이썬은 선택이 아닌 필수! 파이토치로 딥러닝 입문하기!\n\n이 책은 딥러닝 기술...


# 네이버 API로 검색한 결과 최대한 가져오기
* 1회 네이버 서버에 요청
* 응답에서 total을 찾아서 display로 나누어 총 반복할 페이지 산출
* 반복문 작성 및 종료 조건 설정
* start가 1, 101, 201, 301 처럼 변경되어야 함, start의 max값은 1000
* 1페이지 결과를 데이터 프레임으로 만들고 리스트에 append
* 반복이 끝난 후 리스트에 append된 데이터 프레임을 1개로 합침

# for문을 이용해서 전체 페이지 가져오기

In [62]:
keyword = input("검색할 키워드를 입력해 주세요: ")

검색할 키워드를 입력해 주세요: 파이썬


In [76]:
all_data = []

In [77]:
url = "https://openapi.naver.com/v1/search/book.json"
Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
Client_Secret = "by6xHZcrlB"
headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
payload = dict(query=keyword, display=100, start=1, sort="sim")
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
data = r.json()
n_data = data['total']
total_pages = (int(data['total'])//100)+1
print("전체 검색 데이터 수: ", n_data, "검색할 페이지수: ", total_pages )
result_df = json_unpack(data)
all_data.append(result_df)

https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=1&sort=sim
200
전체 검색 데이터 수:  1113 검색할 페이지수:  12


In [88]:
start = 101
for page in range(total_pages):
    print(f"{page+1}/{total_pages} 수집중", end="\r")
    url = "https://openapi.naver.com/v1/search/book.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
    payload = dict(query=keyword, display=100, start=start, sort="sim")
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    data = r.json()
    n_data = data['total']
    total_pages = (int(data['total'])//100)+1
#     print("전체 검색 데이터 수: ", n_data, "검색할 페이지수: ", total_pages )
    result_df = json_unpack(data)
    all_data.append(result_df)
    if start == 1000:
        break
    elif start < 901:
        start += 100
    elif start == 901:
        start += 99

https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=101&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=201&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=301&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=401&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=501&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=601&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=701&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=801&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&

In [89]:
len(all_data)

11

In [91]:
final_result = pd.concat(all_data)
final_result = final_result.reset_index(drop=True)
final_result

,title,link,image,author,discount,publisher,pubdate,isbn,description
0,"혼자 만들면서 공부하는 파이썬 (37개 라이브러리와 API로 보고서, 차트, 게임,...",https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_528593...,문현일,23400,한빛미디어,20250214,9791169213271,“파이썬을 배워서 어디에 활용할 수 있을지 궁금했나요?”\n나의 신용카드 내역 분석...
1,혼자 공부하는 파이썬 (1:1 과외하듯 배우는 프로그래밍 자습서),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_325076...,윤인성,19800,한빛미디어,20220601,9791162245651,혼자 해도 충분하다! 1:1 과외하듯 배우는 파이썬 프로그래밍 자습서\n\n『혼자 ...
2,Do it! 점프 투 파이썬 (중학생도 첫날부터 실습하는 초고속 입문서),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_403540...,박응용,19800,이지스퍼블리싱,20230615,9791163034735,프로그래밍 분야 8년 연속 베스트셀러!\n《Do it! 점프 투 파이썬》 전면 개정...
3,밑바닥부터 시작하는 딥러닝 1(리마스터판) (파이썬으로 익히는 딥러닝 이론과 구현),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_526155...,사이토 고키,23400,한빛미디어,20250124,9791169213387,딥러닝 분야 부동의 베스트셀러!\n머리로 이해하고 손으로 익히는 가장 쉬운 딥러닝 ...
4,50개의 프로젝트로 완성하는 파이썬 업무 자동화 (프로젝트 따라하기로 업무 자동화 ...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_537377...,오토코더,19800,위즈앤북(Wiznbook),20250317,9791198685353,ㆍ퀴즈와 QR 코드 생성부터 나만의 음성 비서 제작까지 다양한 프로젝트 대공개!\n...
...,...,...,...,...,...,...,...,...,...
1095,"챗GPT와 업무자동화 (1분 만에 끝내는 엑셀, 구글 시트, 파워포인트, 파이썬 자...",https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_386135...,김철수,16200,위키북스,20230321,9791158394288,"코드 한 줄 몰라도 1분 만에 끝내는 엑셀, 구글 스프레드시트, 파워포인트 업무 자..."
1096,프로그래밍 정석(개념을 잡아주는) (기초 연산부터 고급 문제해결 능력까지 파이썬을 ...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324972...,제니퍼 캠벨^폴 그리스^제이슨 몬토조^그렉 윌슨,0,에이콘출판,20100504,9788960771338,"『개념을 잡아주는 프로그래밍 정석』은 기본적인 프로그래밍 요소부터 알고리즘, 디버깅..."
1097,머신러닝 딥러닝을 이용한 A.I. 자율주행 RC카 프로젝트 with 파이썬+아두이노...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324665...,서민우,0,앤써북,20220520,9791185553979,〈머신러닝 딥러닝을 이용한 A.I. 자율주행 RC카 프로젝트 with 파이썬+아두이...
1098,파이썬 딥러닝 파이토치(Python Deep Learning PyTorch),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324549...,이경택^방성수^안상준,14800,정보문화사,20201008,9788956748573,파이썬은 선택이 아닌 필수! 파이토치로 딥러닝 입문하기!\n\n이 책은 딥러닝 기술...


# While 문으로 데이터 모두 가져오기

In [1]:
import requests
import pandas as pd

In [2]:
keyword = input("검색 키워드를 입력해 주세요. ")

검색 키워드를 입력해 주세요. 핀테크


In [5]:
all_data2 = []
start = 1

while True:
    url = "https://openapi.naver.com/v1/search/book.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
    payload = dict(query=keyword, display=100, start=start, sort="sim")
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    data = r.json()
    n_data = data['total']
    total_pages = (int(data['total'])//100)+1
    print("전체 검색 데이터 수: ", n_data, "검색할 페이지수: ", total_pages )
    result_df = json_unpack(data)
    all_data2.append(result_df)
    if start == 1000:
        break
    elif start < 901:
        start += 100
    elif start == 901:
        start += 99
    

https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=1&sort=sim
200
전체 검색 데이터 수:  1113 검색할 페이지수:  12
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=101&sort=sim
200
전체 검색 데이터 수:  1113 검색할 페이지수:  12
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=201&sort=sim
200
전체 검색 데이터 수:  1113 검색할 페이지수:  12
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=301&sort=sim
200
전체 검색 데이터 수:  1113 검색할 페이지수:  12
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=401&sort=sim
200
전체 검색 데이터 수:  1113 검색할 페이지수:  12
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=501&sort=sim
200
전체 검색 데이터 수:  1113 검색할 페이지수:  12
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=601&sort=sim
200

In [6]:
len(all_data2)

11

In [9]:
final_result2 = pd.concat(all_data2)
final_result2 = final_result2.reset_index(drop=True)
final_result2

,title,link,image,author,discount,publisher,pubdate,isbn,description
0,"혼자 만들면서 공부하는 파이썬 (37개 라이브러리와 API로 보고서, 차트, 게임,...",https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_528593...,문현일,23400,한빛미디어,20250214,9791169213271,“파이썬을 배워서 어디에 활용할 수 있을지 궁금했나요?”\n나의 신용카드 내역 분석...
1,혼자 공부하는 파이썬 (1:1 과외하듯 배우는 프로그래밍 자습서),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_325076...,윤인성,19800,한빛미디어,20220601,9791162245651,혼자 해도 충분하다! 1:1 과외하듯 배우는 파이썬 프로그래밍 자습서\n\n『혼자 ...
2,Do it! 점프 투 파이썬 (중학생도 첫날부터 실습하는 초고속 입문서),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_403540...,박응용,19800,이지스퍼블리싱,20230615,9791163034735,프로그래밍 분야 8년 연속 베스트셀러!\n《Do it! 점프 투 파이썬》 전면 개정...
3,밑바닥부터 시작하는 딥러닝 1(리마스터판) (파이썬으로 익히는 딥러닝 이론과 구현),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_526155...,사이토 고키,23400,한빛미디어,20250124,9791169213387,딥러닝 분야 부동의 베스트셀러!\n머리로 이해하고 손으로 익히는 가장 쉬운 딥러닝 ...
4,50개의 프로젝트로 완성하는 파이썬 업무 자동화 (프로젝트 따라하기로 업무 자동화 ...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_537377...,오토코더,19800,위즈앤북(Wiznbook),20250317,9791198685353,ㆍ퀴즈와 QR 코드 생성부터 나만의 음성 비서 제작까지 다양한 프로젝트 대공개!\n...
...,...,...,...,...,...,...,...,...,...
1095,"챗GPT와 업무자동화 (1분 만에 끝내는 엑셀, 구글 시트, 파워포인트, 파이썬 자...",https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_386135...,김철수,16200,위키북스,20230321,9791158394288,"코드 한 줄 몰라도 1분 만에 끝내는 엑셀, 구글 스프레드시트, 파워포인트 업무 자..."
1096,프로그래밍 정석(개념을 잡아주는) (기초 연산부터 고급 문제해결 능력까지 파이썬을 ...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324972...,제니퍼 캠벨^폴 그리스^제이슨 몬토조^그렉 윌슨,0,에이콘출판,20100504,9788960771338,"『개념을 잡아주는 프로그래밍 정석』은 기본적인 프로그래밍 요소부터 알고리즘, 디버깅..."
1097,머신러닝 딥러닝을 이용한 A.I. 자율주행 RC카 프로젝트 with 파이썬+아두이노...,https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324665...,서민우,0,앤써북,20220520,9791185553979,〈머신러닝 딥러닝을 이용한 A.I. 자율주행 RC카 프로젝트 with 파이썬+아두이...
1098,파이썬 딥러닝 파이토치(Python Deep Learning PyTorch),https://search.shopping.naver.com/book/catalog...,https://shopping-phinf.pstatic.net/main_324549...,이경택^방성수^안상준,14800,정보문화사,20201008,9788956748573,파이썬은 선택이 아닌 필수! 파이토치로 딥러닝 입문하기!\n\n이 책은 딥러닝 기술...


In [10]:
final_result2.to_csv("네이버검색_책_파이썬.csv", encoding="utf-8-sig", index=False)

# 뉴스, 블로그, 책 한번에 검색해서 저장하기

In [5]:
all_data3 = []

for category in ["blog", "news", "book"]:
    start = 1
    while True:
        url = f"https://openapi.naver.com/v1/search/{category}.json"
        Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
        Client_Secret = "by6xHZcrlB"
        headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
        payload = dict(query=keyword, display=100, start=start, sort="sim")
        r = requests.get(url, params=payload, headers=headers)
        print(r.url)
        print(r.status_code)
        data = r.json()
        n_data = data['total']
        total_pages = (int(data['total'])//100)+1
        print("전체 검색 데이터 수: ", n_data, "검색할 페이지수: ", total_pages )
        result_df = json_unpack(data)
        all_data3.append(result_df)
        if start == 1000:
            break
        elif start < 901:
            start += 100
        elif start == 901:
            start += 99
final_result3 = pd.concat(all_data3)
final_result3 = final_result3.reset_index(drop=True)
final_result3

https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200
전체 검색 데이터 수:  456149 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=201&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=301&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=401&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=501&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=10

,title,link,description,bloggername,bloggerlink,postdate,originallink,pubDate,image,author,discount,publisher,pubdate,isbn
0,<b>핀테크</b> 투자 후기,https://blog.naver.com/kan1687/223812724931,"<b>핀테크</b> 투자, 아직도 망설이고 계신가요? 여러분 혹시 이런 고민 해보...",우정잉♥,blog.naver.com/kan1687,20250329,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,[<b>핀테크</b>] 2024년 <b>핀테크</b> 업계 결산: 놀라운 3가지 변화,https://blog.naver.com/xoom777/223708685044,동안 <b>핀테크</b> 업계에는 예상치 못한 변화의 바람이 불었습니다. 한동안 주...,Ich fange an,blog.naver.com/xoom777,20241229,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,[인사이트 캠퍼스] 2025 동계 <b>핀테크</b> 캠프 (with 전주대학교...,https://blog.naver.com/fininsight/223734638126,전주대학교 친구들이 직접 여의도까지 방문하여 <b>핀테크</b>(FinTech)의 ...,핀인사이트 (Fininsight),blog.naver.com/fininsight,20250122,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,[AI <b>핀테크</b>] AI와 빅데이터로 금융 시장을 재편하는 <b>핀테크</...,https://blog.naver.com/jangpung_/223765345858,이제 대출도 AI가 하는 시대 출처 : 파가야 테크놀러지스 홈페이지 금융과 기술이 ...,낭만을 꿈꾸는 일상,blog.naver.com/jangpung_,20250218,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,IPAY <b>핀테크</b>관련주 ETF -미국 CBDC 코인 서비스 실험 확대,https://blog.naver.com/jaso_/223736810791,IPAY <b>핀테크</b>관련주 ETF -미국 CBDC 코인 서비스 실험 확대 I...,미국주식 퍼주마,blog.naver.com/jaso_,20250130,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2301,"신남방 지역 주요국의 핀테크 발전과 협력방안: 인도, 태국, 말레이시아, 베트남, ...",https://search.shopping.naver.com/book/catalog...,이 책은 대외경제정책연구원의 신남방 지역 주요국의 핀테크 발전과 협력방안을 다룬 정...,NaN,NaN,NaN,NaN,NaN,https://shopping-phinf.pstatic.net/main_332340...,이충열^이종하^이선호,9000,대외경제정책연구원,20201231,9788932290201
2302,[CD] 2018 4차산업 기업.스타트업 주소록 - CD-ROM 1장 (가상현실·드...,https://search.shopping.naver.com/book/catalog...,"2018 4차산업 기업·스타트업 주소록 CD에는 4차 산업기술, 바이오헬스케어, 창...",NaN,NaN,NaN,NaN,NaN,https://shopping-phinf.pstatic.net/main_324666...,한국콘텐츠미디어,170100,한국콘텐츠미디어,20180921,9791155055663
2303,"4차산업 기업·스타트업 주소록(2023)(CD) (IOT, VR, 가상현실, 블록체...",https://search.shopping.naver.com/book/catalog...,"4차산업 기업·스타트업 주소록을 분야별, 지역별로 18,460건, IT기업 상세정보...",NaN,NaN,NaN,NaN,NaN,https://shopping-phinf.pstatic.net/main_400602...,한국콘텐츠미디어 편집부,215100,한국콘텐츠미디어,20230501,8809496290337
2304,"4차산업 기업·스타트업 주소록(2024)(CD) (3D프린터, IOT, VR, 가상...",https://search.shopping.naver.com/book/catalog...,"4차산업 기업·스타트업 주소록을 분야별, 지역별로 19,908건, IT기업 상세정보...",NaN,NaN,NaN,NaN,NaN,https://shopping-phinf.pstatic.net/main_466775...,한국콘텐츠미디어 편집부,215100,한국콘텐츠미디어,20240326,9791172340209
